docs/tutorial/write-your-first-kubernetes-charm-for-a-django-app.rst

```diff
@@ -1,699 +1,781 @@
-    latest/edge`` to get the latest edge version of Charmcraft.
+We will test the Django application by visiting the app in a web
+browser.
 
-MicroK8s is required to deploy the Django application on Kubernetes. Install MicroK8s:
+Change into the ``django_hello_world`` directory:
 
-.. code-block:: bash
+.. code:: bash
 
-    sudo snap install microk8s --channel 1.31-strict/stable
-    sudo adduser $USER snap_microk8s
-    newgrp snap_microk8s
+   cd django_hello_world
 
-Wait for MicroK8s to be ready using ``sudo microk8s status --wait-ready``. Several
-MicroK8s add-ons are required for deployment
+Open the settings file of the application located at
+``django_hello_world/settings.py``. Update the ``ALLOWED_HOSTS`` setting
+to allow all traffic:
 
-.. code-block:: bash
+.. code:: python
 
-    sudo microk8s enable hostpath-storage
-    # Required to host the OCI image of the Django application
-    sudo microk8s enable registry
-    # Required to expose the Django application
-    sudo microk8s enable ingress
+   ALLOWED_HOSTS = ['*']
 
-Juju is required to deploy the Django application. Install Juju and bootstrap a
-development controller:
+Save and close the ``settings.py`` file.
 
-.. code-block:: bash
+Now, run the Django application to verify that it works:
 
-    sudo snap install juju --channel 3.5/stable
-    mkdir -p ~/.local/share
-    juju bootstrap microk8s dev-controller
+.. code:: bash
 
-Finally, create a new directory for this tutorial and go inside it:
+   python3 manage.py runserver 0.0.0.0:8000
 
-.. code-block:: bash
+.. note::
 
-    mkdir django-hello-world
-    cd django-hello-world
+   Specifying ``0.0.0.0:8000`` allows for traffic outside of the Multipass VM.
 
+Now we need the private IP address of the Multipass VM. Outside of the
+Multipass VM, run:
 
-Create the Django application
------------------------------
+.. code-block::
 
-Create a ``requirements.txt`` file, copy the following test into it and then
-save it:
+   multipass info charm-dev | grep IP
 
-.. code-block:: bash
+.. note::
 
-    Django
+   The ``grep`` command extracts a portion of the output to highlight the
+   IP address.
 
-Install ``python3-venv`` and create a virtual environment:
+With the Multipass IP address, we can visit the Django app in a web
+browser. Open a new tab and visit
+``http://<MULTIPASS_PRIVATE_IP>:8000``, replacing
+``<MULTIPASS_PRIVATE_IP>`` with your VM’s private IP address.
 
-.. code-block:: bash
+The Django application should respond in the browser with
+``The install worked successfully! Congratulations!``.
 
-    sudo apt-get update && sudo apt-get install python3-venv -y
-    python3 -m venv .venv
-    source .venv/bin/activate
-    pip install -r requirements.txt
+The Django application looks good, so we can stop it for now from the
+original terminal of the Multipass VM using :kbd:`Ctrl` + :kbd:`C`.
 
-Create a new project using ``django-admin``:
+Pack the Django application into a rock
```

Comment:
Should we link to the [Rockcraft tutorial](https://documentation.ubuntu.com/rockcraft/en/latest/tutorial/flask.html) here as a sort of "see also"? It covers a lot of useful things that are out of scope for the charm tutorial but that a developer really should use before building a real Django charm.

Reply:
I think you and @medubelko should decide that. I'm happy with whatever you two decide!